src/Sarif.Multitool.Library/Rules/RuleId.cs

```diff
@@ -47,6 +47,7 @@ public static class RuleId
         public const string ReviewArraysThatExceedConfigurableDefaults = "GH1004";
         public const string LocationsMustBeRelativeUrisOrFilePaths = "GH1005";
         public const string ProvideCheckoutPath = "GH1006";
+        public const string MessageMustBeFlattened = "GH1007";
```

Comment:
>MessageMustBeFlattened [](http://example.com/codeflow?start=28&length=22)

This rule will tell user the other direction right:
public const string ProvideMessageArguments = "SARIF2002";

will the user get confused

Reply:
It looks like the other names start with the verb when possible. So maybe this on could be "FlattenMessage"

Reply:
Updated to "FlattenResultMessage"

Reply:
right these 2 rules are opposite to each other.
SARIF validation rules are common for all SARIF files.
GH rules only for SARIF files need to be ingested GHAS, should be fine if the GH rule is conflict with SARIF rules. What do you think?

Reply:
:thumbsup:

Reply:
I know that 'flatten' is our special jargon but I wonder if something like 'ProvideFullyFormattedMessageStrings' is more descriptive.